project/stac_api/models.py

```diff
@@ -1,28 +1,307 @@
+    # TODO: overwrite items save() function accordingly
+    # suggestions of fields to be auto-populated:
+    SW = ArrayField(models.FloatField(), blank=True)  # [Float], auto-populated
+    NE = ArrayField(models.FloatField(), blank=True)  # [Float], auto-populated
+
+    collection_name = models.CharField(unique=True, default="")  # string
+    # collection_name is what is simply only called "id" in here:
+    # http://ltboc.infra.bgdi.ch/static/products/data.geo.admin.ch/apitransactional.html#operation/createCollection
+    itemType = models.CharField(default="Feature")  # string
+
+    # The field formerly known as "Keywords" has been "sourced out" to a
+    # separate Keyword class with many-to-many relations to the Collection objects
+
+    license = models.CharField()  # string
+
+    # link objects are now in separate class Link_Collection with many-to-many
+    # relation to Collection objects
+
+    # The field formerly know as "providers" has been "sourced out" to a separate
+    # Provider class with many-to-many relation to the Collection objects.
+
+    # after discussion with Chris and Tobias:
+    # stac_extension will be populated with default values that are set to be
+    # non-editable for the moment. Could be changed, should the need arise.
+    stac_extension = ArrayField(
+        models.CharField(),
+        default=[
+            'eo',
+            'proj',
+            'view',
+            'https://data.geo.admin.ch/stac/geoadmin-extension/1.0/schema.json'
+        ],
+        editable=False
+    )
+    stac_version = models.CharField()  # string
+
+    # "summaries" values will be updated on every update of an item inside the
+    # collection --> TODO: rewrite items save() function accordingly
+    summaries_eo_gsd = models.FloatField(blank=True)
+    summaries_proj = ArrayField(models.IntegerField(blank=True))
+    # after discussion with Chris and Tobias: geoadmin_variant will be an
+    # array field of CharFields. Simple validation is done (e.g. no "Sonderzeichen"
+    # in array)
+    # TODO: Probably geoadmin_variant could be also auto-populated on every
+    # update of an asset --> update the geoadmin_variant of the corresponding
+    # collection
+    geoadmin_variant = ArrayField(models.CharField())
+
+    title = models.CharField(blank=True)  # string
+
+    def __str__(self):
+        return self.collection_name
+
+    def clean(self):
+        # TODO: move this check to the items save()
+        if self.start_date is None and self.end_date is None:
+            raise ValidationError(_('At least a start date or an end date has to be defined.'))
+
+        # very simple validation, raises error when geoadmin_variant strings contain special
+        # characters or umlaut.
+        for variant in self.geoadmin_variant:
+            if not (bool(re.search('^[a-zA-Z0-9]*$', variant)) == True):
+                raise ValidationError(_('Property geoadmin:variant not correctly specified.'))
+
```

Comment:
just for Info: cleaning of individual fields (e.g. the `geoadmin_variant`) can also be done like this
```
def clean_geoadmin_variant(self):
    ....
```

Reply:
Thanks for the info.
For the moment I will stick with the general clean() as I guess there will probably be checks for more than one field per class.
If that will not be the case in the end, we can still change the general clean() to a field specific one.
Or would it make sense to do that from the beginning due to better performance?